src/Common/ThreadStatus.cpp

```diff
@@ -149,7 +149,8 @@ ThreadStatus::~ThreadStatus()
 
     if (deleter)
         deleter();
-    current_thread = nullptr;
+    if (current_thread == this)
```

Comment:
Maybe better to check if `current_thread != this` and throw an exception?

Reply:
No. This was changed because in the case of materialization views, which creates a thread status per view to keep different stats, the ThreadStatus-es will be created and destroyed from the main thread, so current_thread will be equal to the main thread, not the one in the ThreadStatus. Clearly I should leave a comment there.